pkgs/games/openarena/default.nix

```
@@ -1,43 +1,96 @@
   };
 
-  nativeBuildInputs = [ pkgs.unzip patchelf makeWrapper];
-
-  installPhase = let
-    gameDir = "$out/openarena-$version";
-    interpreter = "$(< \"$NIX_CC/nix-support/dynamic-linker\")";
-    libPath = lib.makeLibraryPath [ SDL libglvnd libogg libvorbis curl openal ];
-    arch = {
-      "x86_64-linux" = "x86_64";
-      "i386-linux" = "i386";
-    }.${stdenv.hostPlatform.system};
-  in ''
-    mkdir -pv $out/bin
-    cd $out
-    unzip $src
-
-    patchelf --set-interpreter "${interpreter}" "${gameDir}/openarena.${arch}"
-    patchelf --set-interpreter "${interpreter}" "${gameDir}/oa_ded.${arch}"
-
-    makeWrapper "${gameDir}/openarena.${arch}" "$out/bin/openarena" \
-      --prefix LD_LIBRARY_PATH : "${libPath}"
-    makeWrapper "${gameDir}/oa_ded.${arch}" "$out/bin/oa_ded"
-  '';
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+    curl
+    libglvnd
+    libicns
+    libogg
+    libvorbis
+    makeBinaryWrapper
+    openal
+    patchelf
+    SDL
+  ];
+
+  installPhase =
+    if stdenvNoCC.isDarwin then ''
+      runHook preInstall
+
+      mkdir -p $out/Applications
+      mv $out/*.app $out/Applications
+
+      runHook postInstall
+    '' else
+      let
+        arch = {
+          "x86_64-linux" = "x86_64";
+          "i386-linux" = "i386";
+        }.${stdenvNoCC.hostPlatform.system};
+      in
+      ''
+        runHook preInstall
+
+        mkdir -p $out/share/openarena
+        cp -r {baseoa,missionpack} $out/share/openarena/
+        install openarena.${arch} $out/share/openarena/
+        install oa_ded.${arch} $out/share/openarena/
+
+        icns2png --extract OpenArena.app/Contents/Resources/ioquake3.icns
+
+        for size in 16 32 128 256 512; do
+          mkdir -pv $out/share/icons/hicolor/"$size"x"$size"/apps
+          install -Dm644 ioquake3_"$size"x"$size"x32.png $out/share/icons/hicolor/"$size"x"$size"/apps/openarena.png
+        done;
+
+        makeWrapper "$out/share/openarena/openarena.${arch}" "$out/bin/openarena"
+        makeWrapper "$out/share/openarena/oa_ded.${arch}" "$out/bin/oa_ded"
+
+        runHook postInstall
+      '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "OpenArena";
+      exec = "openarena";
+      icon = "openarena";
+      comment = "A fast-paced 3D first-person shooter, similar to id Software Inc.'s Quake III Arena";
+      desktopName = "openarena";
+      categories = [ "Game" "ActionGame" ];
+    })
+  ];
 
   meta = {
-    description = "Crossplatform openarena client";
+    description = "A fast-paced 3D first-person shooter, similar to id Software Inc.'s Quake III Arena";
     homepage = "http://openarena.ws/";
-    maintainers = [ lib.maintainers.wyvie ];
-    platforms = [ "i386-linux" "x86_64-linux" ];
-    license = lib.licenses.gpl2;
+    license = lib.licenses.gpl2Plus;
+    mainProgram = "openarena";
+    maintainers = with lib.maintainers; [ wyvie ];
+    platforms = [ "i386-linux" "x86_64-linux" ] ++ lib.platforms.darwin;
```

I’m not sure this will even actually work, but the upstream app bundles are for x86 and PPC (no 64-bit builds as far as I can tell).

```suggestion
    platforms = [ "i386-linux" "i686-darwin" "x86_64-linux" ];
```